src/oscar/apps/search/search_handlers.py

```diff
@@ -9,7 +9,43 @@
 FacetMunger = get_class('search.facets', 'FacetMunger')
 
 
-class SearchHandler(object):
+class SearchResultsPaginationMixin(object):
+    paginate_by = None
+    paginator_class = Paginator
+    page_kwarg = 'page'
+
+    def get_paginate_by(self, *args, **kwargs):
+        return self.paginate_by
+
+    def paginate_queryset(self, queryset, page_size):
+        """
+        Paginate the search results. This is a simplified version of
+        Django's MultipleObjectMixin.paginate_queryset
+        """
+        paginator = self.get_paginator(queryset, page_size)
+        page_kwarg = self.page_kwarg
+        page_number = self.request_data.get(page_kwarg, 1)
+        try:
+            page_number = int(page_number)
+        except ValueError:
+            if page_number == 'last':
+                page_number = paginator.num_pages
+            else:
+                raise InvalidPage(_(
+                    "Page is not 'last', nor can it be converted to an int."))
+        # This can also raise an InvalidPage exception.
+        page = paginator.page(page_number)
+        return paginator, page, page.object_list, page.has_other_pages()
```

Comment:
Do we need to add the two new items to the returned tuple, given that both of them are accessible from `page` itself? (Would be nice if we don't have to change the API for projects that are calling this method).

Reply:
Yep agreed. Original intention was to replicate the same behaviour as in the `MultipleObjectMixin.paginate_queryset`. Updated.

Reply:
In fact, no - we still need it, because we inherit both `MultipleObjectsMixin` within this mixin class in the `SimpleProductSearchHandler` and thus this method needs to have the same signature as there.